### ISSUE TITLE
Generic type inference

### ISSUE THREAD

**Issue by md-5** [bradsdavis/candle-decompiler#4]
Since local variables need to have their types inferred, and from that we can infer the type of variables used in each method invocation, it would be possible to recover the generic types of classes / fields removed by obfuscators.

A nice feature to think about in the future.
All the best
md_5

PS: I'll probably be making random tickets slating all the features I think would be a nice addition - and maybe even implement some when the codebase gets a bit further along.

**Comment by bradsdavis** [comment]
Yes, this is something we have planned to try and recover, and is pretty much the same approach we will take.  It will be a least a few months out though.  If you track this comment, I will let you know when I get it in there.  

Keep the feature requests coming!

**Comment by md-5** [comment]
I actually wrote the type inference part for local variables in a separate adapter of mine, so I know its possible :p just got to get around to retyping the efields.

**Comment by bradsdavis** [comment]
Adapter for what, just curious?

**Comment by md-5** [comment]
Well, bytecode adapter, feed bytecode in, get more usable bytecode out.
